version: '3.9'

services:
  server:
    container_name: social_network_server
    build:
      context: .
      dockerfile: server/Dockerfile_server
    ports:
      - "8000:8000"
    restart: always
    env_file:
      - .env
    volumes:
      - ./client/static:/static
    depends_on:
      - postgres
    networks:
      - twitter_network

  postgres:
    container_name: social_network_db
    image: postgres:16
    environment:
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_PASSWORD=${DB_PASS}
    ports:
      - '5432:5432'
    env_file:
      - .env
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${DB_USER}" ]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - twitter_network

  client:
    container_name: social_network_client
    image: nginx
    build:
      context: .
      dockerfile: client/Dockerfile_client
    ports:
      - "80:80"
    volumes:
      - ./client/static:/static
      - ./client/nginx.conf:/etc/nginx/nginx.conf
    depends_on:
      - server
    networks:
      - twitter_network

volumes:
  postgres_data:

networks:
  twitter_network:
    driver: bridge
